WiseBuy/APIs/SearchUPCAPI.m

```diff
@@ -0,0 +1,50 @@
+//
+//  SearchUPCAPI.m
+//  WiseBuy
+//
+//  Created by Mai Ngo on 7/21/22.
+//
+
+#import "SearchUPCAPI.h"
+
+static NSString *const kUpcQuery = @"&upc=";
+static NSString *const kBaseURL = @"https://www.searchupc.com/handlers/upcsearch.ashx?request_type=3&access_token=";
+
+@implementation SearchUPCAPI
+
++ (void)fetchDeals:(nonnull NSString *)barcode {
+    
+    NSString *const path = [[NSBundle mainBundle] pathForResource: @"Keys" ofType: @"plist"];
+    NSDictionary *const dict = [NSDictionary dictionaryWithContentsOfFile: path];
+    NSString *accessToken = [dict objectForKey: @"access_Token"];
+    NSString *const accessTokenStr = accessToken;
+    
+    if ([[NSUserDefaults standardUserDefaults] stringForKey:@"access_Token"]) {
+        accessToken = [[NSUserDefaults standardUserDefaults] stringForKey:@"access_Token"];
+    }
+    
+    NSString *fullURL = [NSString stringWithFormat:@"%@%@%@%@", kBaseURL, accessTokenStr, kUpcQuery, barcode];
+    
+    NSString *userKey = [dict objectForKey: @"searchUPC_userKey"];
+    
+    if ([[NSUserDefaults standardUserDefaults] stringForKey:@"searchUPC_userKey"]) {
+        userKey = [[NSUserDefaults standardUserDefaults] stringForKey:@"searchUPC_userKey"];
+    }
+    NSDictionary *headers = @{
+        @"user_key": userKey
```

Comment:
Is there a chance that userKey is nil? If so, this code will crash

Reply:
ah i will check for this